File: rust/src/models/list_response_message_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

use super::message_out::MessageOut;

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct ListResponseMessageOut {
    pub data: Vec<MessageOut>,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub iterator: Option<String>,

    #[serde(rename = "prevIterator")]
    #[serde(skip_serializing_if = "Option::is_none")]
    pub prev_iterator: Option<String>,

    pub done: bool,
}

impl ListResponseMessageOut {
    pub fn new(data: Vec<MessageOut>, done: bool) -> Self {
        Self {
            data,
            iterator: None,
            prev_iterator: None,
            done,
        }
    }
}

File: csharp/Svix/Models/ListResponseMessageOut.cs
// this file is @generated
using Newtonsoft.Json;

namespace Svix.Models
{
    public class ListResponseMessageOut : BaseModel
    {
        [JsonProperty("data", Required = Required.Always)]
        public required List<MessageOut> Data { get; set; }

        [JsonProperty("done", Required = Required.Always)]
        public required bool Done { get; set; }

        [JsonProperty("iterator")]
        public string? Iterator { get; set; } = null;

        [JsonProperty("prevIterator")]
        public string? PrevIterator { get; set; } = null;
    }
}
